influxdb3_write/src/last_cache/table_function.rs

```diff
@@ -0,0 +1,93 @@
+use std::{any::Any, sync::Arc};
+
+use arrow::datatypes::SchemaRef;
+use async_trait::async_trait;
+use datafusion::{
+    common::{plan_err, Result},
+    datasource::{function::TableFunctionImpl, TableProvider, TableType},
+    execution::context::SessionState,
+    logical_expr::{Expr, TableProviderFilterPushDown},
+    physical_plan::{memory::MemoryExec, ExecutionPlan},
+    scalar::ScalarValue,
+};
+
+use super::{LastCache, LastCacheProvider};
+
+#[async_trait]
+impl TableProvider for LastCache {
```

Comment:
this is pretty cool -- I think we could totally support something like this in distributed (the key would be, of course, to get and populate `LastCacheProvider` correctly in a distributed setting, which is likely pretty compilcated

Reply:
Perfect use case for Redis 😄